src/infrastructure/repository/ride/MongoRidesRepository.js

```diff
@@ -0,0 +1,38 @@
+const RideMapper = require('./MongoRideMapper');
+
+
+class MongoRideRepository {
+
+  constructor({ DbRideModel }) {
+    this.DbRideModel = DbRideModel;
+  }
+
+  // RULES IN AS ENTITY --- OUT AS ENTITY
+  async add(ride) {
+    // Validate and check user existance throw error if any
+    await this._validate(ride);
+
+    //1. Before save to database conver data from entity structure to database structure using same folder mapper
+    const DbRide = new this.DbRideModel(RideMapper.toDatabase(ride));
+    const newDbRide = await DbRide.save();
+
+    // 2 After getting the newly saved date from database convert the result back as enitity
+    return RideMapper.toEntity(newDbRide);
+
+  }
+
+  //Private functions
+  async _validate(ride) {

```

Comment:
This function does not use object property. For be (I think) cleaner and totally private maybe extract it in a "real" function out of the object. Moreover this approach make the function totally stateless for object.